psydac/fem/vector.py

```diff
@@ -356,6 +389,9 @@ def get_refined_space(self, ncells):
         return self._refined_space[tuple(ncells)]
 
     def set_refined_space(self, ncells, new_space):
+        # [YG, 27.03.2025]: It appears that this method is assuming that the
+        # refined space has ldim=2, and that the number of cells is the same
+        # along each axis. These two conditions are very strong.
```

Comment:
good point 👍

Reply:
This will raise an error immediately if we try to use it in 3D, or with different number of cells in each direction. Better than failing silently or giving an undefined behavior, I guess! 😅

Reply:
Absolutely. The comment will be useful to clarify the error when it pops up